greeting.rb

```diff
@@ -0,0 +1,42 @@
+class Greeting
+  def greet(*name)
+    if name.any?
+      low_case = name.reject { |x| x == x.upcase }
+      up_case= name.select { |r| r == r.upcase }
+      up_case_string = up_case.join(",")
+      name_string = name.join("")
+      if name.length == 1 and name == low_case
+        "Hello, #{name.join(",")}."
+      elsif name.length == 1 and name == up_case
+        "HELLO #{name.join(",")}!"
+      elsif name.length == 2 and !name_string.include?(",") and !name_string.include?("\"")
+        "Hello, #{name.first} and #{name.last}."
+      elsif name.length > 2 and name == low_case
+        "Hello, #{name[0..-2].join(", ")}, and #{name.last}."
+      elsif name.length > 2 and name.include? up_case_string
```

Comment:
Instead of using this `up_case_string`, why not just ask if `up_case.any?`? That is one less variable to know and reason about.